drivers/java/src/main/java/com/rethinkdb/net/Cursor.java

```diff
@@ -93,12 +99,17 @@ protected void maybeSendContinue() {
         }
     }
 
-    protected void waitOnCursorItems(){
+    protected void waitOnCursorItems(Optional<Long> timeout) throws TimeoutException {
         Response res = null;
         try {
-            res = this.awatingContinue.get();
-        }
-        catch(Exception e){
+            if(timeout.isPresent()){
+                res = this.awatingContinue.get(timeout.get(), TimeUnit.MILLISECONDS);
+            } else {
+                res = this.awatingContinue.get();
+            }
+        }catch(TimeoutException exc){
```

Comment:
Since you're rethrowing this and it's in the signature, this exception block can be skipped